### ISSUE TITLE
Race condition signal: killed

### ISSUE THREAD

**Issue by mattymo** [roboll/helmfile#1687]
Possibly duplicate of https://github.com/roboll/helmfile/issues/1166

I have a k8s cluster that runs CI for several components and there can be up to 50 parallel helmfile commands running across 8 nodes. Here's the failure I'm seeing during the execution:

```
FAILED RELEASES:
NAME
mkt
in product/helmfile.yaml: failed processing release mkt: command "/usr/bin/helm" exited with non-zero status:

PATH:
  /usr/bin/helm

ARGS:
  0: helm (4 bytes)
  1: upgrade (7 bytes)
  2: --install (9 bytes)
  3: --reset-values (14 bytes)
  4: mkt (3 bytes)
  5: stable/mkt (10 bytes)
  6: --wait (6 bytes)
  7: --timeout (9 bytes)
  8: 1500s (5 bytes)
  9: --create-namespace (18 bytes)
  10: --namespace (11 bytes)
  11: mkt-60-4b7 (10 bytes)
  12: --values (8 bytes)
  13: /tmp/values386558221 (20 bytes)
  14: --values (8 bytes)
  15: /tmp/values491133192 (20 bytes)
  16: --values (8 bytes)
  17: /tmp/values623731911 (20 bytes)
  18: --history-max (13 bytes)
  19: 10 (2 bytes)

ERROR:
  signal: killed

EXIT STATUS
  -1
```

I should stress that each container has limits of 3 vcpu and 2gb memory. The helmfile that I use has a dependency chain and installs in 5 phases: 2 charts, 3 charts, 2 charts, 3 charts, then finally 7 charts. this "mkt" chart is number 6 out of 7 of the last group. 

Additionally there are no process limits and max open files is 1048576.

I'm running a relatively old version of helmfile. I'll try an upgrade to see if anything changes
helmfile v0.130.0
helm v3.5.2

**Comment by mattymo** [comment]
@mumoshu Any chance you can help out here?

**Comment by mumoshu** [comment]
@mattymo Hey! Sorry I had been blown away by the big waves of github notifications.

First of all, is this caused by a race condition? At glance it looks like your helm process is getting OOM killed due to helmfile trying to run dozens of helm processes in parallel. If that's your issue, you can just set lower concurrency like `helmfile --concurrency 5 apply`

**Comment by mattymo** [comment]
Lowering concurrency helped

**Comment by mumoshu** [comment]
Not sure why I didn't come up with this idea but I believe we'd better default to a smaller concurrency, perhaps to the number of cores, rather than the number of releases defined in a said helmfile.yaml which can be extremely high.

**Comment by skizot722** [comment]
Hello! Is it possible to re-open this issue?  We're encountering the same problem.  We have 121 releases (some with deps on others), being executed on a 16-core / 64 GB RAM VM, with `--concurrency 16`.  It seems to happen almost always 3 minutes into execution.  But we are able to get a successful run after enough retries.  The behavior feels like a race condition, as the author of this issue alluded to.  Would you be able to take a further look at things @mumoshu ?

**Comment by yxxhero** [comment]
Please post a new issue in helmfile/helmfile
